[Brian Cheong] iP

SKCheongBrian opened the pull request:
# BriBot
> "Let our advance worrying become advance thinking and planning." - Winston Churchill ([source](https://www.brainyquote.com/quotes/winston_churchill_156920))

Bribot helps you manage your tasks so that you have less to worry about. It's,

- CLI just run it in your command line or terminal!
- very **_intuitive_** 😉
- easy to get up running

We are able to track,

1. Todos for things that you just need to complete.
2. Deadlines for things that need to be completed by a certain date and time.
3. Events for events that occur at a certain date and time.

- [x] Managing tasks.
- [ ] Managing reminders.

---

Look at the revolutionary Java `main` method used to run the program.

```Java
Public static void main(String[] args) {
    new Duke("./src/main/data/tasks.txt").run();
}
```

Commit:
Add Gradle support

SKCheongBrian committed:
Implemented Level-1: starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

SKCheongBrian committed:
added ability to add items and list them out. As well as fixed input bug

SKCheongBrian committed:
completed level-3: add done feature

SKCheongBrian committed:
Completed Level-4: add support for tracking Todos, Deadlines and Events.

SKCheongBrian committed:
Testing using the I/O redirection technique

SKCheongBrian committed:
Added error handling

SKCheongBrian committed:
Fixed bugs and changed tasks from linked list to ArrayList

SKCheongBrian committed:
Added delete task feature

SKCheongBrian committed:
Added use of enum

SKCheongBrian committed:
Fix code style issues

SKCheongBrian committed:
Add feature to copy tasks into a text file

SKCheongBrian committed:
Add feature to be able to read from text file to get tasks from previous session

SKCheongBrian committed:
Add actual date time representation for Deadlines and Events

SKCheongBrian committed:
Merge branch 'branch-Level-8'

SKCheongBrian committed:
Fix bugs caused by the merge from level 7 and level 8

SKCheongBrian committed:
Refactor code to be more OOP

SKCheongBrian committed:
Refactor program into packages

SKCheongBrian committed:
Add JUnit testing

SKCheongBrian committed:
Remove extra print statement

SKCheongBrian committed:
Add javadocs

SKCheongBrian committed:
Add find feature

SKCheongBrian committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/TaskList.java

SKCheongBrian committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branc-A-Gradle

SKCheongBrian committed:
Add find method in TaskList that was somehow missing

SKCheongBrian committed:
Automate project builds using Gradle

SKCheongBrian committed:
Add checkstyle and corrected styling errors

Linxcathyyy commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,141 @@
+     * @return An ArrayList of tasks that match the given input.
+     */
+    public ArrayList<Task> find(String input) {
+        ArrayList<Task> res = new ArrayList<>();
```
What does res mean here? Perhaps a more intuitive variable name here? ☺️

Linxcathyyy reviewed (commented):
Good job! Your codes follow the coding standard quite well! ☺️ Try to name your variables more intuitive in the future for better readability~

SKCheongBrian committed:
Add gui

SKCheongBrian committed:
Add FXML

SKCheongBrian committed:
Add bot greeting

Kangxun commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,57 @@
+/**
+ * Represents an add command where it is given a task to add to the tasklist
+ */
+
```
Maybe delete this newline for readability?

Kangxun commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * @param ui the ui of the program.
+     * @param storage the storage of the program.
+     * @throws DukeException thrown if there is error when trying to add command.
+     */
```
Perhaps include a `@return` here to help with understanding of the code?

Kangxun commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.UnknownCommand;
```
i like the separation of commands into specific classes 👍🏻

Kangxun commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,114 @@
+            String[] words = input.split(" ", 2);
+            String keyWord = words[0];
+            String rest;
+            String[] splitRest;
```
Perhaps change the naming to a term that could be less confused as a verb such as `restOfString`?

Kangxun reviewed (commented):
Great work in following naming conventions and best practices! Only minor adjustments are noted in the files. Awesome work 💯

SKCheongBrian committed:
Fix initialisation within Duke.java

Because it is good practice

Add the this keyword for the creation of the fields in a duke object

SKCheongBrian committed:
Add assertions to check the length of the commands

Currently there is no assertions in the program

There should be assertions in the program

Add assertions to check the length of the command

SKCheongBrian committed:
Refactor code to fit checkstyle and have good code quality

Is failing the checkstyle

Fix code to follow checkstyle

To follow checkstyle

SKCheongBrian committed:
Merge pull request #1 from SKCheongBrian/branch-A-Assertions

Add assertions to check the length of the commands

SKCheongBrian committed:
Fix UI bug

SKCheongBrian committed:
Merge branch 'master' into branch-A-CodeQuality

SKCheongBrian committed:
Merge pull request #2 from SKCheongBrian/branch-A-CodeQuality

Refactor code to fit checkstyle and have good code quality

SKCheongBrian committed:
Add sort function

No sort function for the list of tasks

Tasks do not appear chronologically in order of their date and timing

Makes more sense to appear in chronological order

SKCheongBrian committed:
Revert "Add sort function"

This reverts commit 3d93c66a98291757fa57ee6629ed182d1774106f.

SKCheongBrian committed:
Revert "Revert "Add sort function""

This reverts commit b91dd55548df0975cea2ef0fde65db1c34aa216a.

SKCheongBrian committed:
Merge branch 'branch-A-Assertions'

# Conflicts:
#	src/main/resources/view/DialogBox.fxml

SKCheongBrian committed:
Add better gui

SKCheongBrian committed:
Add user guide and UI screenshot

SKCheongBrian committed:
Set theme jekyll-theme-minimal

SKCheongBrian committed:
Change name to bribot

SKCheongBrian committed:
Merge branch 'master' of https://github.com/SKCheongBrian/ip

SKCheongBrian committed:
Fix bugs and improved code quality
